daemon/execdriver/windows/run.go

```diff
@@ -105,18 +110,30 @@ func (d *Driver) Run(c *execdriver.Command, pipes *execdriver.Pipes, hooks execd
 		HostName:                c.Hostname,
 	}
 
-	for i := 0; i < len(c.LayerPaths); i++ {
-		_, filename := filepath.Split(c.LayerPaths[i])
+	for _, layerPath := range c.LayerPaths {
+		_, filename := filepath.Split(layerPath)
 		g, err := hcsshim.NameToGuid(filename)
 		if err != nil {
 			return execdriver.ExitStatus{ExitCode: -1}, err
 		}
 		cu.Layers = append(cu.Layers, layer{
 			ID:   g.ToString(),
-			Path: c.LayerPaths[i],
+			Path: layerPath,
 		})
 	}
 
+	// Add the mounts (volumes, bind mounts etc) to the structure
+	if len(c.Mounts) > 0 {
+		var mds []mappedDir
```

Comment:
@jhowardmsft 

`mds := make([]mappedDir, 0, len(c.Mounts))` would be preferable since you know in advance how much memory to allocate.

That, or:
```Go
mds := make([]mappedDir, len(c.Mounts))
for i, mount := range c.Mounts {
  mds[i] = mappedDir{ ... }
}
```

I don't have a strong preference between the two forms.